scylla-cql/src/errors.rs

```diff
@@ -40,6 +40,10 @@ pub enum QueryError {
 
     #[error("Unable to allocate stream id")]
     UnableToAllocStreamId,
+
+    /// Client timeout occurred before any response arrived
+    #[error("Request timeout: {0}")]
+    RequestTimeout(String),
```

Comment:
Hmm, it looks like we will have two "timeout" errors now... Maybe we could provide a helper method `is_timeout()` which tells whether this error is a timeout (either a client- or server-side timeout)?